### PR TITLE
[KeyVault] adding soft-delete methods and models for storage entities to 7.0

### PR DESCRIPTION
**Pull request by schaabs**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by rijethma — CHANGES_REQUESTED** [review]
added my comments

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/examples/recoverStorageSasDefinition.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+         "headers":{
+
+         },
+         "body":{
+            "id":"https://storage-sdk-test.vault-int.azure-int.net/storage/recoversas01/sas/recoverStrgSasDef01",
+            "sid":"https://storage-sdk-test.vault-int.azure-int.net/secrets/recoversas01-recoverStrgSasDef01",
+            "templateUri":"se=2018-02-01T00%3A00Z&spr=https&sv=2017-04-17&sr=b&sig=XXFNfuMCHYrBx0bhemJ7PWn0xGfImMXT6LfbXWvtRUk%3D",
+            "sasType":"service",
+            "validityPeriod":"P15D",
+            "attributes":{
```
Comment (dragav) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/examples/recoverStorageSasDefinition.json]: The recoveryLevel attribute seems to be missing; is the service returning it?
[...]
I just checked the service code, and it seems that the controller is indeed setting the attribute. The recoveryLevel should be present with every response that returns the entity's attributes.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/keyvault.json [hunk 2]
```diff
@@ -4606,6 +4981,24 @@
...
+          "type": "string",
+          "readOnly": true,
+          "description": "The URL to get the next set of deleted storage accounts."
+        }
+      },
+      "description": "The deleted storage account list result"
+    },
     "SasDefinitionAttributes": {
       "properties": {
         "enabled": {
```
[the remark this answers is not in the document]
Comment (schaabs) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/keyvault.json]: Added

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/keyvault.json [hunk 3]
```diff
@@ -4723,6 +5168,24 @@
...
+          "readOnly": true,
+          "description": "A response message containing a list of the deleted SAS definitions in the vault along with a link to the next page of deleted sas definitions"
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The URL to get the next set of deleted storage accounts."
+        }
+      },
+      "description": "The deleted storage account list result"
```
[the remark this answers is not in the document]
Comment (schaabs) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/keyvault.json]: I updated the descriptions on ListSasDefinitionsResult, but GH seems to have been confused by the diff and moved this comment.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/keyvault.json [hunk 4]
```diff
@@ -2569,6 +2569,91 @@
...
+          },
+          "default": {
+            "description": "Key Vault error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/KeyVaultError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Recover Deleted Storage Account": {
```
[the remark this answers is not in the document]
Comment (schaabs) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/keyvault.json]: This was changed but for some reason the updated review didn't mark the comment as outdated.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/examples/recoverStorageSasDefinition.json (hunk 1 above)
Comment (schaabs) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.0/examples/recoverStorageSasDefinition.json]: Updated all the examples which return StorageAccountAttributes or SasDefinitionAttributes to include recoverLevel in the response.

**Review by dragav — APPROVED** [review]

**Review by rijethma — APPROVED** [review]

**Comment by sergey-shandar** [comment]
@schaabs Please have a look the CI jobs https://travis-ci.org/Azure/azure-rest-api-specs/jobs/339114359 and https://travis-ci.org/Azure/azure-rest-api-specs/jobs/339114358

**Pull request merged**